source/test/TestApp/PlayFabEventTest.cpp

```diff
@@ -218,6 +220,56 @@ namespace PlayFabUnit
         std::bind(&PlayFabEventTest::NonStaticEmitEventCallback, this, std::placeholders::_1, std::placeholders::_2));
     }
 
+    void PlayFabEventTest::GenericMultiThreadedTest(uint32_t pNumThreads, uint32_t pNumEventsPerThread)
+    {
+        uint32_t localNumEventsReturned = 0;
+        uint32_t localTotalEventsExpected = pNumThreads * pNumEventsPerThread;
+        for (uint32_t thread = 0; thread < pNumThreads; ++thread)
+        {
+            testThreadPool.emplace_back([&localNumEventsReturned, localTotalEventsExpected, pNumEventsPerThread, this]() {
+                std::shared_ptr<PlayFabEventAPI*> api = SetupEventTest();
+                for (uint32_t i = 0; i < pNumEventsPerThread; ++i)
+                {
+                    (*api)->EmitEvent(MakeEvent(0, PlayFabEventType::Default),
+                    [&localNumEventsReturned, pNumEventsPerThread, localTotalEventsExpected]
+                    (std::shared_ptr<const PlayFab::IPlayFabEvent>, std::shared_ptr<const PlayFab::IPlayFabEmitEventResponse>)
+                    {
+                        localNumEventsReturned++;
```

Comment:
>localNumEventsReturned [](start = 24, length = 22)

every thread has access to this value by reference. access to it needs to be properly synchronized to prevent concurrent writes/reads